crates/polars-ops/src/chunked_array/list/namespace.rs

```diff
@@ -342,8 +342,18 @@ pub trait ListNameSpaceImpl: AsList {
     /// So index `0` would return the first item of every sublist
     /// and index `-1` would return the last item of every sublist
     /// if an index is out of bounds, it will return a `None`.
-    fn lst_get(&self, idx: i64) -> PolarsResult<Series> {
+    fn lst_get(&self, idx: i64, null_on_oob: bool) -> PolarsResult<Series> {
         let ca = self.as_list();
+        if !null_on_oob
+            && ca.iter().any(|sublist| {
```

Comment:
Iterating the lists is very expensive to do a bound check. We instead instead could create an iterator that returns the lenghts of the `ListArray's`. The `ListArray` has an `OffsetBuffer` which in turn has `lengths` iterator.